src/sage/graphs/base/sparse_graph.pyx

```diff
@@ -631,21 +663,12 @@ cdef class SparseGraph(CGraph):
 
             Don't forget to free ``p_pointers[0]``  !
         """
-        cdef int num_nbrs = 0
         cdef int degree = self.out_degrees[u]
         if degree == 0:
             p_pointers[0] = NULL
             return 0
-        cdef SparseGraphBTNode **pointers = <SparseGraphBTNode **>check_allocarray(degree, sizeof(SparseGraphBTNode *))
-        p_pointers[0] = pointers
-
-        cdef SparseGraphBTNode* v = self.next_out_neighbor_BTNode_unsafe(u, -1)
-        while v:
-            pointers[num_nbrs] = v
-            num_nbrs += 1
-            v = self.next_out_neighbor_BTNode_unsafe(u, v.vertex)
-
-        return num_nbrs
+        p_pointers[0] = <SparseGraphBTNode **>check_allocarray(degree, sizeof(SparseGraphBTNode *))
```

Comment:
here you allocate some memory but I think it is never freed. But may be it is done elsewhere...

Reply:
The code was in the unused functions that are now deleted.
(For info, the code was written like this because the old functions assumed that an unallocated pointer was passed as argument, that the pointer was allocated inside the function and should be freed after the call.)